backend/src/apiserver/resource/resource_manager.go

```diff
@@ -1241,15 +1242,21 @@ func (r *ResourceManager) GetPipelineVersionTemplate(versionId string) ([]byte,
 	// Verify pipeline version exist
 	_, err := r.pipelineStore.GetPipelineVersion(versionId)
 	if err != nil {
-		return nil, util.Wrap(err, "Get pipeline version template failed")
+		return nil, util.Wrap(err, "Get pipeline version template failed: cannot get pipeline version")
 	}
 
 	template, err := r.objectStore.GetFile(r.objectStore.GetPipelineKey(fmt.Sprint(versionId)))
 	if err != nil {
-		return nil, util.Wrap(err, "Get pipeline version template failed")
+		return nil, util.Wrap(err, "Get pipeline version template failed: cannot get file")
 	}
 
-	return template, nil
+	// convert template from JSON to YAML
+	template_yaml, err := yaml.JSONToYAML(template)
+	if err != nil {
+		return nil, util.Wrap(err, "Get pipeline version template failed: unable to convert template from JSON to YAML")
+	}
```

Comment:
Can we add some tests for validating this PR change?

1. If pipeline template was stored as JSON format, return a JSON format template.
2. If pipeline template was stored as YAML format, return a YAML format template.

Reply:
That's a great idea! I will do it.

Reply:
That's a great idea! I will do it.